modules/openapi-generator/src/main/resources/python/api.mustache

```diff
@@ -35,307 +36,203 @@ class {{classname}}:
+    {{#asyncio}}async {{/asyncio}}def {{operationId}}{{>partial_api_args}} -> {{{returnType}}}{{^returnType}}None{{/returnType}}:
+{{>partial_api}}
+        response_data = {{#asyncio}}await {{/asyncio}}self.api_client.call_api(
+            *_param,
+            _request_timeout=_request_timeout
+        )
+        {{#asyncio}}await {{/asyncio}}response_data.read()
+        return self.api_client.response_deserialize(
+            response_data=response_data,
+            response_types_map=_response_types_map,
+        ).data
 
-{{#notes}}
-        {{{.}}}  # noqa: E501
-{{/notes}}
-{{^asyncio}}
-        This method makes a synchronous HTTP request by default. To make an
-        asynchronous HTTP request, please pass async_req=True
 
-        >>> thread = api.{{operationId}}({{#allParams}}{{paramName}}, {{/allParams}}async_req=True)
-        >>> result = thread.get()
-{{/asyncio}}
+    @validate_call
+    {{#asyncio}}async {{/asyncio}}def {{operationId}}_with_http_info{{>partial_api_args}} -> ApiResponse[{{{returnType}}}{{^returnType}}None{{/returnType}}]:
+{{>partial_api}}
+        response_data = {{#asyncio}}await {{/asyncio}}self.api_client.call_api(
+            *_param,
+            _request_timeout=_request_timeout
+        )
+        {{#asyncio}}await {{/asyncio}}response_data.read()
+        return self.api_client.response_deserialize(
+            response_data=response_data,
+            response_types_map=_response_types_map,
+        )
 
-{{#allParams}}
-        :param {{paramName}}:{{#description}} {{{.}}}{{/description}}{{#required}} (required){{/required}}{{#optional}}(optional){{/optional}}
-        :type {{paramName}}: {{dataType}}{{#optional}}, optional{{/optional}}
-{{/allParams}}
-{{^asyncio}}
-        :param async_req: Whether to execute the request asynchronously.
-        :type async_req: bool, optional
-{{/asyncio}}
-        :param _request_timeout: timeout setting for this request.
-               If one number provided, it will be total request
-               timeout. It can also be a pair (tuple) of
-               (connection, read) timeouts.
-        :return: Returns the result object.
-                 If the method is called asynchronously,
-                 returns the request thread.
-        :rtype: {{returnType}}{{^returnType}}None{{/returnType}}
-        """
-        kwargs['_return_http_data_only'] = True
-        if '_preload_content' in kwargs:
-            message = "Error! Please call the {{operationId}}_with_http_info method with `_preload_content` instead and obtain raw data from ApiResponse.raw_data"  # noqa: E501
-            raise ValueError(message)
 
-        return {{#asyncio}}await {{/asyncio}}self.{{operationId}}_with_http_info.raw_function(
-            {{#allParams}}
-            {{paramName}},
-            {{/allParams}}
-            **kwargs,
+    @validate_call
+    {{#asyncio}}async {{/asyncio}}def {{operationId}}_without_preload_content{{>partial_api_args}} -> RESTResponseType:
```

Comment:
Every generated client will only ever support one backend library at a time. That's why I'm not sure the `RESTResponseType` alias is useful in the final generated client. 

For me, not using async or tornado, seeing `urllib3.HTTPResponse` here would be most informative when working with the generated client.

Reply:
I agree with @robertschweizer : I don't think there's a need to abstract the underlying response from the actual client.

The client itself is already a bit abstracted because it's doing internal manipulation of the parameters but the response type doesn't do anything special, and at best it would be a downgraded version of the underlying response.

Since it's being used only in specific cases (when the caller only wants to do the HTTP call without the model API), and that it's not possible anyway to use multiple HTTP clients at the same, I would just remove the abstraction and return the HTTP réponse object from the underlying library.

Reply:
Separating the generated client(urllib3/aiohttp) from the api(`api.mustache`) will eliminate the need to modify `api.mustache` when adding a new `RESTClientObject`.

My idea is to be able to add a new RESTClientObject with minimal changes.
(In the future I may add httpx as a client.)

Reply:
Ok I see. Re-reading the change, we can maybe try to remove `RESTReponse` instead, but keep the type alias `RESTReponseType` for the use case you mentioned.

This can be done in a subsequent PR though.